binary_tree_from_preorder_inorder.py

```diff
@@ -0,0 +1,34 @@
+'''
+Time complexity: O(2n*n)
+Space complexity: O(max_depth)
```

Comment:
Since you are creating sub_arrays as well, I believe your SC would be bigger than this.
You can do way better than O(n^2) TC. Please improve your solution.